parcels/kernel.py

```diff
@@ -96,24 +99,51 @@ def __init__(self, fieldset, ptype, pyfunc=None, funcname=None,
 
             basename = path.join(get_cache_dir(), self._cache_key)
             self.src_file = "%s.c" % basename
-            self.lib_file = "%s.so" % basename
+            self.lib_file = "%s.%s" % (basename, 'dll' if platform == 'win32' else 'so')
             self.log_file = "%s.log" % basename
         self._lib = None
 
+    def __del__(self):
+        # Clean-up the in-memory dynamic linked libraries.
+        # This is not really necessary, as these programs are not that large, but with the new random
+        # naming scheme which is required on Windows OS'es to deal with updates to a Parcels' kernel.
+        if not self._lib == None:
```

Comment:
In Parcels, we're following (most) of the PEP8 formatting rules. This is what flake8 checks on Travis, and it keeps our code clean and easy-to-read. According to https://travis-ci.org/OceanParcels/parcels/builds/299949591?utm_source=github_status&utm_medium=notification, there are a few minor PEP8 violations, including here, where it's better to write `if self._lib is not None`